### DISCUSSION TITLE
why every time connect must switch to eth ?

### DISCUSSION THREAD

**Discussion by callmejm** [wevm/wagmi#1985] [General]
I have [mainnet, bsc, arbitrum] in configureChains, every time when I try to connect wallet, while my MetaMask already connected to Arbitrum, but it still force me to switch to Ethereum network. I using [web3modal](https://github.com/WalletConnect/web3modal), but I also have a try on [connectkit](https://docs.family.co/connectkit/try-it-out) still forcing me to go Ethereum first

```
const { chains, provider } = configureChains(
  [mainnet, bsc, arbitrum],
  [
    walletConnectProvider({ projectId: walletconnectProjectID as string}),
    jsonRpcProvider({
      rpc: (chain) => {
        return { http: chain.rpcUrls.default, webSocket: undefined } as any
      },
      priority: 1
    }),
    alchemyProvider({apiKey: alchemyAPIKEY as string, priority: 0}),
  ]
);

const wagmiClient = createClient({
  autoConnect: true,
  connectors: modalConnectors({
    appName: 'MY PROJECT',
    version: "1",
    chains,
    projectId: walletconnectProjectID as string
  }),
  provider
})
```

**Comment by tmm** [comment]
wagmi doesn't have this behavior by default. Must be something on ConnectKit and Web3Modal's side.

**Reply by callmejm** [reply]
but both are using wagmi

**Reply by tmm** [reply]
Yes, they must of added this behavior as part of their code.

**Reply by kozerog555** [reply]
When you think the Arbitrum will launch the token?
